### PR TITLE
Properly sort named nodes, blank nodes and unbounded values

### LINKED ISSUE #892: ORDER BY is not implemented completely
_This issue has been modified by @rubensworks to describe the intended solution_

Apparently, [we only support ordering by literal values](https://github.com/comunica/sparqlee/blob/7c1180638154c72d3c3292e48d000ed9770796f0/lib/util/Ordering.ts#L8), and a total ordering as mentioned in the [SPARQL spec](https://www.w3.org/TR/sparql11-query/#modOrderBy) has not been implemented yet.

The goal of this issue is to properly implement total ordering in sparqlee (with the necessary unit tests), and optionally some integration tests here in Comunica.

-----

#### Issue type:

- :bug: Bug 

____
#### Description:

Running queries with ORDER BY on @comunica/actor-init-sparql-file, IRIs are not sorted.

Relevant paragraph in the SPARQL 1.1 Spec (https://www.w3.org/TR/sparql11-query/#modOrderBy):

> The "<" operator (see the Operator Mapping and 17.3.1 Operator Extensibility) defines the relative order of pairs of numerics, simple literals, xsd:strings, xsd:booleans and xsd:dateTimes. Pairs of IRIs are ordered by comparing them as simple literals.

Example (using VALUES for self-containment, but I've found the same issue with actual input sources):
```javascript
import { newEngine } from '@comunica/actor-init-sparql-file'

const query = `
  PREFIX ex: <http://example.org/>
  SELECT ?x
  WHERE {
    VALUES ?x {ex:b ex:a ex:c}
  } ORDER BY ?x`;

const engine = newEngine();
const resPromise = engine.query(query);

// show the results
resPromise
  .then((result) => result.bindings())
  .then((ba) => ba.map((b) => Object.fromEntries(b.entries())))
  .then(console.log);

// expected output:
//
// [
//   {
//     '?x': NamedNode { termType: 'NamedNode', value: 'http://example.org/a' }
//   },
//   {
//     '?x': NamedNode { termType: 'NamedNode', value: 'http://example.org/b' }
//   },
//   {
//     '?x': NamedNode { termType: 'NamedNode', value: 'http://example.org/c' }
//   }
// ]

// actual output:
//
// [
//   {
//     '?x': NamedNode { termType: 'NamedNode', value: 'http://example.org/b' }
//   },
//   {
//     '?x': NamedNode { termType: 'NamedNode', value: 'http://example.org/a' }
//   },
//   {
//     '?x': NamedNode { termType: 'NamedNode', value: 'http://example.org/c' }
//   }
// ]
```

____
#### Environment:

Result of running from my package `node node_modules/@comunica/actor-init-sparql/bin/query.js -v`

| software            | version
| ------------------- | -------
| Comunica Init Actor | 1.22.3
| node                | v14.18.1
| npm                 | 6.14.15
| yarn                | 1.22.17
| Operating System    | linux (Linux 5.4.0-48-generic)

#### Crash log:

Not applicable.

-----

## Bounty

A [bounty](https://comunica.dev/association/bounties/) has been placed on this issue by:

| |
|-|
| [Comunica Association](https://comunica.dev/association/) |
| €544 |

[Click here to learn more](https://comunica.dev/association/bounty_process/) if you're interested in claiming this bounty by resolving this issue.

### PR DESCRIPTION
**Pull request by Tpt**
The sorting order according to the SPARQL spec should be undefined < blank node < named node < literal

Closes #892

### CODE REVIEW

File: packages/actor-query-operation-orderby-sparqlee/lib/ActorQueryOperationOrderBySparqlee.ts [hunk 1]
```diff
@@ -52,8 +52,8 @@ export class ActorQueryOperationOrderBySparqlee extends ActorQueryOperationTyped
       const transform = async(bindings: Bindings, next: any, push: (result: IAnnotatedBinding) => void):
       Promise<void> => {
         try {
-          const result = await evaluator.evaluate(bindings);
-          push({ bindings, result });
+          const result = await evaluator.evaluateAsInternal(bindings);
```
Comment (rubensworks) [inline packages/actor-query-operation-orderby-sparqlee/lib/ActorQueryOperationOrderBySparqlee.ts:55]: If `toRDF` is undefined, isn't this a problem in sparqlee?

I guess we should be able to keep using `evaluate`.

**Review by rubensworks — CHANGES_REQUESTED** [review]
Excellent! Just one comment regarding `evaluateAsInternal`

File: packages/actor-query-operation-orderby-sparqlee/lib/ActorQueryOperationOrderBySparqlee.ts (hunk 1 above)
Comment (rubensworks) [reply packages/actor-query-operation-orderby-sparqlee/lib/ActorQueryOperationOrderBySparqlee.ts]: Perfect! :-)

I just published sparqlee 2.1.0. Could you update the PR to use that version? Will merge right away after that.

**Review by rubensworks — APPROVED** [review]

File: packages/actor-query-operation-orderby-sparqlee/lib/ActorQueryOperationOrderBySparqlee.ts (hunk 1 above)
Comment (Tpt) [reply packages/actor-query-operation-orderby-sparqlee/lib/ActorQueryOperationOrderBySparqlee.ts]: Thank you! I have just done it.

**Pull request merged**